src/libraries/System.Runtime/tests/System.Globalization.Calendars.Tests/ISOWeek/ISOWeekTests.cs

```diff
@@ -132,6 +165,30 @@ public static void ToDateTime_WithInvalidDayOfWeek_Throws(int dayOfWeek)
             AssertExtensions.Throws<ArgumentOutOfRangeException>(nameof(dayOfWeek), () => ISOWeek.ToDateTime(2018, 1, (DayOfWeek)dayOfWeek));
         }
 
+        [Theory]
+        [InlineData(0)]
+        [InlineData(10000)]
+        public static void ToDateOnly_WithInvalidYear_Throws(int year)
+        {
+            AssertExtensions.Throws<ArgumentOutOfRangeException>(nameof(year), () => ISOWeek.ToDateOnly(year, 1, DayOfWeek.Friday));
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(54)]
+        public static void ToDateOnly_WithInvalidWeek_Throws(int week)
+        {
+            AssertExtensions.Throws<ArgumentOutOfRangeException>(nameof(week), () => ISOWeek.ToDateOnly(2018, week, DayOfWeek.Friday));
+        }
+
+        [Theory]
+        [InlineData(-1)]
+        [InlineData(8)]
+        public static void ToDateOnly_WithInvalidDayOfWeek_Throws(int dayOfWeek)
+        {
+            AssertExtensions.Throws<ArgumentOutOfRangeException>(nameof(dayOfWeek), () => ISOWeek.ToDateOnly(2018, 1, (DayOfWeek)dayOfWeek));
+        }
+
```

Comment:
Can these be combined with the already existing methods?

Reply:
Calling both the ToDateTime and ToDateOnly checks in a single test method should be possible. I will adjust the tests to combine these.